backend/routes/friends.js

```diff
@@ -5,24 +5,62 @@ var cors = require('cors');
 
 router.use(cors())
 
-router.get('/users/:name', async (req, res) => {
+/**
+ * find user based on username search and filter out people you're already friends with
+ */
+router.get('/users', async (req, res) => {
     try {
+        let sessionQuery = new Parse.Query("_Session").equalTo("sessionToken", req.query.sessionToken)
+        let userId = await sessionQuery.first({useMasterKey : true})
+        userId = userId.attributes.user.id
         let query = new Parse.Query("_User")
-        query.equalTo("username", req.params.name)
+        query.equalTo("username", req.query.name)
+        query.notEqualTo("objectId", userId)
+
+        //to exclude existing friends or already sent friend requests
+        let query1 = new Parse.Query("Friends").equalTo("fromName", req.query.name).equalTo("toUser", userId)
+        let query2 = new Parse.Query("Friends").equalTo("toName", req.query.name).equalTo("fromUser", userId)
+        const existingFriendsQuery = Parse.Query.or(query1, query2)
+        let existingFriends = await existingFriendsQuery.first()
+        if (existingFriends) {
+            //if there already is a friendship:
+            query.notContainedIn("objectId", [existingFriends.get("fromUser"), existingFriends.get("toUser")])
+        }
         const user = await query.first({useMasterKey : true})
         res.status(200).send(user)
     } catch (err) {
         res.status(400).send({"error" : err })
     }
 })
 
+/**
+ * get list of all pending friend requests that have been sent to a user
+ */
+router.get('/seeReqs/:sessionToken', async(req, res) => {
+    try {
+        let query = new Parse.Query("_Session")
+        query.equalTo("sessionToken", req.params.sessionToken)
+        let user = await query.first({useMasterKey : true})
+        user = user.attributes.user.id
```

Comment:
The friend relationship table could be simplified. We can discuss this more on our 1:1